### ISSUE TITLE
bad pip-develop interaction

### ISSUE THREAD

**Issue by martindurant** [conda/conda#9795]
### Steps to Reproduce

```
conda create -n bare python -c defaults -y
conda activate bare
conda config --set pip_interop_enabled true
pip install -e .  # installs fsspec 0.7.1+3.gafc7cc4
conda install intake -c defaults
```
gives
```
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/common/logic.py", line 126, in _convert
        return self.names[name]
    KeyError: 'defaults/noarch::fsspec-0.6.3-py_0'

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/exceptions.py", line 1079, in __call__
        return func(*args, **kwargs)
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/cli/main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
        return getattr(module, func_name)(args, parser)
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/cli/main_install.py", line 20, in execute
        install(args, parser, 'install')
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/cli/install.py", line 265, in install
        should_retry_solve=(_should_retry_unfrozen or repodata_fn != repodata_fns[-1]),
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/core/solve.py", line 117, in solve_for_transaction
        should_retry_solve)
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/core/solve.py", line 158, in solve_for_diff
        force_remove, should_retry_solve)
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/core/solve.py", line 275, in solve_final_state
        ssc = self._add_specs(ssc)
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/core/solve.py", line 558, in _add_specs
        (_.to_match_spec() for _ in ssc.prefix_data.iter_records()))), self.specs_to_add
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/resolve.py", line 1105, in get_conflicting_specs
        C = r2.gen_clauses()
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/resolve.py", line 909, in gen_clauses
        nkey = C.Not(self.to_sat_name(prec))
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/common/logic.py", line 145, in Not
        return self._eval(self._clauses.Not, (x,), (), polarity, name)
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/common/logic.py", line 131, in _eval
        args = self._convert(args)
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/common/logic.py", line 121, in _convert
        return type(x)(map(self._convert, x))
      File "/Users/mdurant/anaconda/lib/python3.7/site-packages/conda/common/logic.py", line 128, in _convert
        raise ValueError("Unregistered SAT variable name: {}".format(name))
    ValueError: Unregistered SAT variable name: defaults/noarch::fsspec-0.6.3-py_0

`$ /Users/mdurant/anaconda/bin/conda install -c defaults intake`

  environment variables:
                 CIO_TEST=<not set>
        CONDA_DEFAULT_ENV=bare
                CONDA_EXE=/Users/mdurant/anaconda/bin/conda
             CONDA_PREFIX=/Users/mdurant/anaconda/envs/bare
           CONDA_PREFIX_1=/Users/mdurant/anaconda
           CONDA_PREFIX_2=/Users/mdurant/anaconda/envs/py3
    CONDA_PROMPT_MODIFIER=(bare)
         CONDA_PYTHON_EXE=/Users/mdurant/anaconda/bin/python
               CONDA_ROOT=/Users/mdurant/anaconda
              CONDA_SHLVL=3
                     PATH=/Users/mdurant/anaconda/bin:/Users/mdurant/anaconda/envs/bare/bin:/Use
                          rs/mdurant/anaconda/condabin:/Users/mdurant/.cargo/bin:/Users/mdurant/
                          Downloads/google-cloud-sdk/bin:/Users/mdurant/anaconda/envs/py3/bin:/U
                          sers/mdurant/anaconda/condabin:/Users/mdurant/.cargo/bin:/Users/mduran
                          t/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin
               PYTHONPATH=/Users/mdurant/python/
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : bare
    active env location : /Users/mdurant/anaconda/envs/bare
            shell level : 3
       user config file : /Users/mdurant/.condarc
 populated config files : /Users/mdurant/.condarc
          conda version : 4.8.3
    conda-build version : 3.18.11
         python version : 3.7.3.final.0
       virtual packages : __osx=10.13.6
       base environment : /Users/mdurant/anaconda  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /Users/mdurant/anaconda/pkgs
                          /Users/mdurant/.conda/pkgs
       envs directories : /Users/mdurant/anaconda/envs
                          /Users/mdurant/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.3 requests/2.23.0 CPython/3.7.3 Darwin/17.7.0 OSX/10.13.6
                UID:GID : 501:20
             netrc file : None
           offline mode : False


```
(does not happen without pip interop, but then my dev packages tend to get overwritten)

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> /Users/mdurant/.condarc <==
auto_update_conda: False
pip_interop_enabled: False
ssl_verify: True
channel_priority: strict
channels:
  - conda-forge
  - defaults

```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at /Users/mdurant/anaconda/envs/bare:
#
# Name                    Version                   Build  Channel
ca-certificates           2020.1.1                      0    defaults
certifi                   2019.11.28               py38_1    defaults
fsspec                    0.7.1+3.gafc7cc4           dev_0    <develop>
libcxx                    4.0.1                hcfea43d_1    defaults
libcxxabi                 4.0.1                hcfea43d_1    defaults
libedit                   3.1.20181209         hb402a30_0    defaults
libffi                    3.2.1                h475c297_4    defaults
ncurses                   6.2                  h0a44026_0    defaults
openssl                   1.1.1e               h1de35cc_0    defaults
pip                       20.0.2                   py38_1    defaults
python                    3.8.2                hc70fcce_0    defaults
readline                  8.0                  h1de35cc_0    defaults
setuptools                46.1.1                   py38_0    defaults
sqlite                    3.31.1               ha441bb4_0    defaults
tk                        8.6.8                ha441bb4_0    defaults
wheel                     0.34.2                   py38_0    defaults
xz                        5.2.4                h1de35cc_4    defaults
zlib                      1.2.11               h1de35cc_3    defaults

```
</p></details>

**Comment by martindurant** [comment]
ping here, any thoughts? I do not have a decent dev experience, because my local dev-installed pip packages depend on one-another.

**Comment by kalefranz** [comment]
I just ran into this yesterday too.

**Comment by kalefranz** [comment]
It's definitely a bug, and an unpleasant one. Introduced most likely somewhere in 4.7 or 4.8. First step will be to bisect versions and find out where.

**Comment by BjornFJohansson** [comment]
I got this as well. I can not install anything if I have some 
package installed by pip develop.

**Comment by martindurant** [comment]
What became of `conda develop` ??

**Comment by thomco** [comment]
Any update on this?  I'm experiencing the exact same stack.  I'm stuck in the water

**Comment by slamer59** [comment]
Hello, same here with `conda-forge/linux-64::webencodings-0.5-py35_0`

**Comment by cjmartian** [comment]
@martindurant I was able to recreate your issue with conda 4.8.3

![image](https://user-images.githubusercontent.com/47338163/87975209-eee04d00-ca90-11ea-9721-aab765abf318.png)

However, when I try to reproduce the same issue from master, I do not get the same result:

![image](https://user-images.githubusercontent.com/47338163/87975294-0b7c8500-ca91-11ea-8dc1-b24a35ac6927.png)

![image](https://user-images.githubusercontent.com/47338163/87975439-41216e00-ca91-11ea-8011-776e3eb97861.png)

![image](https://user-images.githubusercontent.com/47338163/87975398-32d35200-ca91-11ea-86e5-cceef2a563d1.png)

I can look into what changed, but it looks like master seems to be working.
If I completely missed something, please let me know.

**Comment by elexira** [comment]
still not working for me

**Comment by Mortimerp9** [comment]
@elexira, I confirm, I changed `pip_interop_enabled` to be `false` and this error went away.

**Comment by Titus-von-Koeller** [comment]
Hi @cjmartian,

I was also getting this error, but in my case it could be solved by deleting the environment and installing it from scratch.

I hope that my report will be useful in any case.

In contrast to the others in this thread, I had no luck with `pip_interop_enabled: false`. The error persisted with the prior `false`-flag as well as when not having it in my `~/.condarc` at all. So, this doesn't seem to be a "solution" for everyone running into this specific ValueError.

I was running into this with the `conda-lock` library, but it seems this issue is a `conda` bug: [see related issue](https://github.com/conda/conda-pack/issues/107).

I attached the logs of the error here: [conda_error.log](https://github.com/conda/conda/files/9459845/conda_error.log)

```❯ conda --version               
conda 4.13.0
```

I just tried updating conda and the error persists with `conda 4.14.0`.

```❯ conda config --show-sources
==> /home/nelius/.condarc <==
channel_priority: strict
channels:
  - conda-forge
```
